src/digitalIdentity-frontend/src/app/pages/credential/create-creDef/create-creDef.component.ts

```diff
@@ -0,0 +1,293 @@
+    name: new FormControl('', [Validators.required]),
+    comment: new FormControl(''),
+    revocable: new FormControl(false),
+    iconUrl: new FormControl(''),
+    schemaId: new FormControl(''),
+  });
+
+  creDefTmp: Credential = {
+    name: '',
+    comment: '',
+    imageUri: '',
+    revocable: false,
+    schemaId: '',
+    image: null,
+  };
+  creDef: Credential = {
+    name: '',
+    comment: '',
+    imageUri: '',
+    revocable: false,
+    schemaId: '',
+    image: null,
+  };
+  error = '';
+  fileName = '';
+
+  constructor(
+    private fb: FormBuilder,
+    private http: HttpClient,
+    private dialogRef: MatDialog,
+    private router: Router,
+    private httpService: BackendHttpService
+  ) {
+    //this.creDefFormGroup =
+  }
+
+  ngOnInit(): void {
+    this.getSchema();
+    this.schemaCtrl.setValue(this.schemaData[this.schemaData.length]);
+
+    // load the initial creDef list
+    this.filteredSchemas.next(this.schemaData.slice());
+
+    // listen for search field value changes
+    this.schemaFilterCtrl.valueChanges
+      .pipe(takeUntil(this._onDestroy))
+      .subscribe(() => {
+        this.filtered_Schemas();
+      });
+  }
+
+  ngAfterViewInit() {
+    this.setInitialValue();
+  }
+
+  ngOnDestroy() {
+    this._onDestroy.next();
+    this._onDestroy.complete();
+  }
+
+  setInitialValue() {
+    this.filteredSchemas
+      .pipe(take(1), takeUntil(this._onDestroy))
+      .subscribe(() => {
+        this.singleSelect.compareWith = (
+          a: schemaDataType,
+          b: schemaDataType
+        ) => a && b && a.alias === b.alias;
+      });
+  }
+
+  filtered_Schemas() {
+    if (!this.schemaData) {
+      return;
+    }
+    // get the search keyword
+    let search = this.schemaFilterCtrl.value;
+    if (!search) {
+      this.filteredSchemas.next(this.schemaData.slice());
+      return;
+    } else {
+      search = search.toLowerCase();
+    }
+    // filter the schemas
+    this.filteredSchemas.next(
+      this.schemaData.filter(
+        (schemaData) => schemaData.alias.toLowerCase().indexOf(search) > -1
+      )
+    );
+  }
+
+  selectedSchema = '';
+
+  onSelected(event: any): void {
+    this.selectedSchema = event.value;
+    this.creDef.schemaId = this.selectedSchema;
+  }
+
+  selectFile(event: any) {
+    if (!event.target.files[0] || event.target.files[0].length == 0) {
+      this.error = 'You must select an image';
+      return;
+    }
+
+    let fileType = event.target.files[0].type;
+
+    console.log(fileType);
+    console.log(event.target.files[0]);
+
+    if (fileType.match(/image\/*/) == null) {
+      this.error = 'Only images are supported';
+      return;
+    }
+
+    this.fileName = event.target.files[0].name;
+
+    let reader = new FileReader();
+    reader.readAsDataURL(event.target.files[0]);
+
+    reader.onload = (_event) => {
+      this.error = '';
+      this.creDefFormGroup.controls['iconUrl'].setValue(reader.result); //the uploaded image is here
```

Comment:
You are setting only the imageUri/iconUrl here, not the image itself. It works, because lissi only requires the imageUri and not the image file (this one is optional), but later, you are adding the image (file) (which is empty) to the http params. Please refer also the real image file to lissi :)